src/pybind/mgr/cephadm/nfs.py

```diff
@@ -0,0 +1,128 @@
+import logging
+import rados
+
+from typing import Dict, Optional
+
+from ceph.deployment.service_spec import NFSServiceSpec
+
+import cephadm
+from orchestrator import OrchestratorError
+
+from . import utils
+
+logger = logging.getLogger(__name__)
+
+class NFSGanesha(object):
+    def __init__(self,
+                 mgr,
+                 daemon_id,
+                 spec):
+        # type: (cephadm.CephadmOrchestrator, str, NFSServiceSpec) -> None
+        self.mgr = mgr
+        self.daemon_id = daemon_id
+        self.spec = spec
+
+    def get_rados_user(self):
+        # type: () -> str
+        return '%s.%s' % (self.spec.service_type, self.daemon_id)
+
+    def get_rados_config_name(self):
+        # type: () -> str
+        return 'conf-' + self.get_rados_user()
+
+    def get_rados_config_url(self):
+        # type: () -> str
+        url = 'rados://' + self.spec.pool + '/'
+        if self.spec.namespace:
+            url += self.spec.namespace + '/'
+        url += self.get_rados_config_name()
+        return url
```

Comment:
As I mentioned to @varshar16 in an email, the original design had each daemon watching its own config file for notifications that it should reload. Given that in the current design, all of them will eventually include the same common config file, we could have them all watch that file instead, and then you'd only need to send a single notify to make all the daemons reload.

Note that this may have implications for existing orch drivers (e.g. rook), so this may need careful consideration.

What may make the most sense is to move the notification code into the orchestrator drivers themselves. That way if an orchestrator backend has a different watch_urls scheme you could paper over it. Maybe a new "notify_config_change" method in the orchestrator or something?

Reply:
> What may make the most sense is to move the notification code into the orchestrator drivers themselves. That way if an orchestrator backend has a different watch_urls scheme you could paper over it. Maybe a new "notify_config_change" method in the orchestrator or something?

I think that is the intent behind the `orch reconfig` action, but it will tear down each container and then spin up new ones, whereas the RADOS notify + SIGHUP seems more efficient.

Reply:
Right.  We can make reconfig do the notify (or sighup, or whatever)...